### DISCUSSION TITLE
v3.2.0

### DISCUSSION THREAD

**Discussion by danielroe** [nuxt/nuxt#18915] [Announcements]
> **3.2.0** is the first minor release since we've started our new release schedule. We've brought it forward by a couple of weeks to include some goodies we want you to be able to play with soon.

## 👀 Highlights

- ⚡️ **Nuxt DevTools**
   > You can opt-in to Nuxt DevTools per-project by going to the project root and running:
   >
   > ```bash
   > npx nuxi@latest devtools enable
   > ```
   >
   > Restart your Nuxt server and open your app in browser. Click the Nuxt icon on the bottom (or press `Alt+D`) to toggle the DevTools.
   
   More information in the [docs](https://github.com/nuxt/devtools)!
- ✨ **Better DX for overriding** `runtimeConfig`, including inline type helpers <br><br><img src="https://user-images.githubusercontent.com/28706372/215903089-9f071c5f-50a1-45cd-841d-173f3b5e66db.png" width="65%">
- 🪄 **Automatically inferred return type** for `useFetch` and `$fetch` based on method.

   > It'll be a type error to use the wrong method when hitting an endpoint.
   >
   > Plus, if you have multiple methods served by a single endpoint (like `~/server/api/test.get.ts` and `~/server/api/test.post.ts` then the response type will match the kind of response you make.
- 🍪 `useFetch` is now integrated with `event.$fetch`, meaning **cookies and context are now passed to api requests automagically** within internal requests.
- 🔥 We now **treeshake client-only components** out of the server build more effectively using the experimental `treeshakeClientOnly` feature

   > This is turned on by default but if you experience any issues, you can turn this off via:
   > ```js
   > export default defineNuxtConfig({
   >   experimental: {
   >     treeshakeClientOnly: false
   >   }
   > })
   > ```
- 🛠️ New **`addRouteMiddleware` kit utility** for module authors
- 💪 **Nitropack v2.2** has been released
   > Lots of features, including runtime proxy support using route rules, nested fetch calls, binary and raw storage operations, exposed `event.context.cf` (cloudflare) and built-in session support.
   >
   > For full details see [release notes](https://github.com/unjs/nitro/blob/main/CHANGELOG.md#v220)

## Changelog

[compare changes](https://github.com/nuxt/nuxt/compare/v3.1.2...v3.2.0)
[release](https://github.com/nuxt/nuxt/releases/tag/v3.2.0)

**Comment by madebyfabian** [comment]
Exciting stuff! Thanks to the team for their great work 💚

**Comment by DrJume** [comment]
Sadly, some PR references under the changelog section in the release (mostly ~#9000) are linking to wrong repo nuxt/nuxt when they should be linking to their correct PRs in nuxt/framework.

**Reply by DrJume** [reply]
Or the commit messages were not updated to link to the PRs which were copied from nuxt/framework to nuxt/nuxt?

**Comment by georgiai1** [comment]
The link to Dev Tools docs doesn't work

**Reply by danielroe** [reply]
It's now public...

**Comment by oleghalin** [comment]
Having issue with useAsyncComponent after upgrade 
<img width="581" alt="image" src="https://user-images.githubusercontent.com/9086983/217943986-10488263-f794-4499-b4ef-c2a5c5f15446.png">
<img width="669" alt="image" src="https://user-images.githubusercontent.com/9086983/217944006-c1b14ea0-6a95-49b2-a3e2-c27d17d08ae2.png">


And have next error:

```
[[vite-node] [plugin:vite:import-analysis] [VITE_ERROR] ./components/folder/component.vue:26:5 <br><pre>24 |      Object.defineProperty(__returned__, "__isScriptSetup", { enumerable: false, value: true });
25 |      return __returned__;
26 |    }
   |      ^
27 |  });
28 |  import { withCtx as _withCtx, toDisplayString as _toDisplayString, createTextVNode as _createTextVNode, resolveComponent as _resolveComponent, createVNode as _createVNode, mergeProps as _mergeProps } from "vue";</pre><br>
at ./components/folder/component.vue:26:5 ] {
  statusCode: 500,
  fatal: false,
  unhandled: false,
  statusMessage: 'Vite Error',
  __nuxt_error: true
```

If delete <client-only> component mounts, but i get hydration issue for sure

**Reply by oleghalin** [reply]
While previous version works correctly

**Reply by danielroe** [reply]
You can try disabling `experimental.treeShakeClientOnly` - but would you open a new issue with a [reproduction](https://nuxt.com/docs/community/reporting-bugs/#create-a-minimal-reproduction)? 🙏

**Reply by oleghalin** [reply]
Yea, it works, i will create repro today

**Reply by oleghalin** [reply]
https://github.com/nuxt/nuxt/issues/18949

**Comment by michaelsynan** [comment]
Many thanks and respect to the Nuxt team!
